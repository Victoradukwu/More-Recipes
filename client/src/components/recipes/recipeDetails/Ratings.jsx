import React from 'react';
import PropTypes from 'prop-types';

const Ratings = ({ recipe }) => (
  <div>
    <br /><br />

    <div className="card bg-light text-black px-0 text-center">
      <img className="card-img" src={require('../../../assets/img/lg5.jpg')} alt="Sample recipe" style={{ height: '250px' }} />
      <div className="card-body">
        <div>
          <span className="fa fa-thumbs-up fa-lg " style={{ float: 'left' }}>{recipe.upvote}</span>
          <span className="fa fa-star fa-lg" style={{ float: 'none' }}>{recipe.favorites}</span>
          <span className="fa fa-thumbs-down fa-lg" style={{ float: 'right' }}>{recipe.downvote}</span>
        </div>
      </div>
    </div>
  </div>

);
Ratings.propTypes = {
  recipe: PropTypes.object.isRequired
};
export default Ratings;
