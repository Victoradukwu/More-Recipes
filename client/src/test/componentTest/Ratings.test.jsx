import React from 'react';
import { shallow } from 'enzyme';
import Ratings from '../../components/recipes/recipeDetails/Ratings';

const props = {
  recipe: { recipeName: 'theRecipe', id: 1 },
  upvoteRecipe: jest.fn(),
  downvoteRecipe: jest.fn(),
  favoriteRecipe: jest.fn()
};
describe('Ratings', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Ratings {...props} />);
    expect(wrapper.find('div').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls upvoteRecipe function', () => {
    const wrapper = shallow(<Ratings {...props} />);
    const upvoteBtn = wrapper.find('.fa-thumbs-up');
    upvoteBtn.simulate('click');
    expect(props.upvoteRecipe).toHaveBeenCalled();
  });

  it('calls downvoteRecipe function', () => {
    const wrapper = shallow(<Ratings {...props} />);
    const downvoteBtn = wrapper.find('.fa-thumbs-down');
    downvoteBtn.simulate('click');
    expect(props.downvoteRecipe).toHaveBeenCalled();
  });

  it('calls favorite Recipe function', () => {
    const wrapper = shallow(<Ratings {...props} />);
    const favBtn = wrapper.find('.fa-heart');
    favBtn.simulate('click');
    expect(props.favoriteRecipe).toHaveBeenCalled();
  });
});
