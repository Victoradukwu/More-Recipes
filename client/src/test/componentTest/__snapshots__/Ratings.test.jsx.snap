// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings takes snapshot matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ratings
    downvoteRecipe={[MockFunction]}
    favoriteRecipe={[MockFunction]}
    recipe={
      Object {
        "id": 1,
        "recipeName": "theRecipe",
      }
    }
    upvoteRecipe={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <br />,
        <br />,
        <div
          className="card bg-light text-black px-0 text-center"
          id="ratings"
        >
          <img
            alt="Sample recipe"
            className="card-img"
            src="http://res.cloudinary.com/victoradukwu/image/upload/v1520509962/hd7.jpg"
            style={
              Object {
                "height": "400px",
              }
            }
          />
          <div
            className="card-body"
          >
            <div>
              <button
                className=" btn btn-lg fa fa-thumbs-up fa-lg "
                id="upvote"
                onClick={[Function]}
                style={
                  Object {
                    "float": "left",
                  }
                }
              >
                   
              </button>
              <button
                className=" btn btn-lg fa fa-heart fa-lg"
                id="favorite"
                onClick={[Function]}
                style={
                  Object {
                    "float": "none",
                  }
                }
              />
              <button
                className="btn btn-lg fa fa-thumbs-down fa-lg"
                id="downvote"
                onClick={[Function]}
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                 
              </button>
            </div>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Sample recipe"
              className="card-img"
              src="http://res.cloudinary.com/victoradukwu/image/upload/v1520509962/hd7.jpg"
              style={
                Object {
                  "height": "400px",
                }
              }
            />,
            <div
              className="card-body"
            >
              <div>
                <button
                  className=" btn btn-lg fa fa-thumbs-up fa-lg "
                  id="upvote"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "left",
                    }
                  }
                >
                     
                </button>
                <button
                  className=" btn btn-lg fa fa-heart fa-lg"
                  id="favorite"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "none",
                    }
                  }
                />
                <button
                  className="btn btn-lg fa fa-thumbs-down fa-lg"
                  id="downvote"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                    }
                  }
                >
                   
                </button>
              </div>
            </div>,
          ],
          "className": "card bg-light text-black px-0 text-center",
          "id": "ratings",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Sample recipe",
              "className": "card-img",
              "src": "http://res.cloudinary.com/victoradukwu/image/upload/v1520509962/hd7.jpg",
              "style": Object {
                "height": "400px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                <button
                  className=" btn btn-lg fa fa-thumbs-up fa-lg "
                  id="upvote"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "left",
                    }
                  }
                >
                     
                </button>
                <button
                  className=" btn btn-lg fa fa-heart fa-lg"
                  id="favorite"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "none",
                    }
                  }
                />
                <button
                  className="btn btn-lg fa fa-thumbs-down fa-lg"
                  id="downvote"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                    }
                  }
                >
                   
                </button>
              </div>,
              "className": "card-body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className=" btn btn-lg fa fa-thumbs-up fa-lg "
                    id="upvote"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "left",
                      }
                    }
                  >
                       
                  </button>,
                  <button
                    className=" btn btn-lg fa fa-heart fa-lg"
                    id="favorite"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "none",
                      }
                    }
                  />,
                  <button
                    className="btn btn-lg fa fa-thumbs-down fa-lg"
                    id="downvote"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "right",
                      }
                    }
                  >
                     
                  </button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "   ",
                      undefined,
                    ],
                    "className": " btn btn-lg fa fa-thumbs-up fa-lg ",
                    "id": "upvote",
                    "onClick": [Function],
                    "style": Object {
                      "float": "left",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    "   ",
                    undefined,
                  ],
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": " btn btn-lg fa fa-heart fa-lg",
                    "id": "favorite",
                    "onClick": [Function],
                    "style": Object {
                      "float": "none",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      undefined,
                    ],
                    "className": "btn btn-lg fa fa-thumbs-down fa-lg",
                    "id": "downvote",
                    "onClick": [Function],
                    "style": Object {
                      "float": "right",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    undefined,
                  ],
                  "type": "button",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <br />,
          <br />,
          <div
            className="card bg-light text-black px-0 text-center"
            id="ratings"
          >
            <img
              alt="Sample recipe"
              className="card-img"
              src="http://res.cloudinary.com/victoradukwu/image/upload/v1520509962/hd7.jpg"
              style={
                Object {
                  "height": "400px",
                }
              }
            />
            <div
              className="card-body"
            >
              <div>
                <button
                  className=" btn btn-lg fa fa-thumbs-up fa-lg "
                  id="upvote"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "left",
                    }
                  }
                >
                     
                </button>
                <button
                  className=" btn btn-lg fa fa-heart fa-lg"
                  id="favorite"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "none",
                    }
                  }
                />
                <button
                  className="btn btn-lg fa fa-thumbs-down fa-lg"
                  id="downvote"
                  onClick={[Function]}
                  style={
                    Object {
                      "float": "right",
                    }
                  }
                >
                   
                </button>
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Sample recipe"
                className="card-img"
                src="http://res.cloudinary.com/victoradukwu/image/upload/v1520509962/hd7.jpg"
                style={
                  Object {
                    "height": "400px",
                  }
                }
              />,
              <div
                className="card-body"
              >
                <div>
                  <button
                    className=" btn btn-lg fa fa-thumbs-up fa-lg "
                    id="upvote"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "left",
                      }
                    }
                  >
                       
                  </button>
                  <button
                    className=" btn btn-lg fa fa-heart fa-lg"
                    id="favorite"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "none",
                      }
                    }
                  />
                  <button
                    className="btn btn-lg fa fa-thumbs-down fa-lg"
                    id="downvote"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "right",
                      }
                    }
                  >
                     
                  </button>
                </div>
              </div>,
            ],
            "className": "card bg-light text-black px-0 text-center",
            "id": "ratings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Sample recipe",
                "className": "card-img",
                "src": "http://res.cloudinary.com/victoradukwu/image/upload/v1520509962/hd7.jpg",
                "style": Object {
                  "height": "400px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  <button
                    className=" btn btn-lg fa fa-thumbs-up fa-lg "
                    id="upvote"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "left",
                      }
                    }
                  >
                       
                  </button>
                  <button
                    className=" btn btn-lg fa fa-heart fa-lg"
                    id="favorite"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "none",
                      }
                    }
                  />
                  <button
                    className="btn btn-lg fa fa-thumbs-down fa-lg"
                    id="downvote"
                    onClick={[Function]}
                    style={
                      Object {
                        "float": "right",
                      }
                    }
                  >
                     
                  </button>
                </div>,
                "className": "card-body",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className=" btn btn-lg fa fa-thumbs-up fa-lg "
                      id="upvote"
                      onClick={[Function]}
                      style={
                        Object {
                          "float": "left",
                        }
                      }
                    >
                         
                    </button>,
                    <button
                      className=" btn btn-lg fa fa-heart fa-lg"
                      id="favorite"
                      onClick={[Function]}
                      style={
                        Object {
                          "float": "none",
                        }
                      }
                    />,
                    <button
                      className="btn btn-lg fa fa-thumbs-down fa-lg"
                      id="downvote"
                      onClick={[Function]}
                      style={
                        Object {
                          "float": "right",
                        }
                      }
                    >
                       
                    </button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "   ",
                        undefined,
                      ],
                      "className": " btn btn-lg fa fa-thumbs-up fa-lg ",
                      "id": "upvote",
                      "onClick": [Function],
                      "style": Object {
                        "float": "left",
                      },
                    },
                    "ref": null,
                    "rendered": Array [
                      "   ",
                      undefined,
                    ],
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": " btn btn-lg fa fa-heart fa-lg",
                      "id": "favorite",
                      "onClick": [Function],
                      "style": Object {
                        "float": "none",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        undefined,
                      ],
                      "className": "btn btn-lg fa fa-thumbs-down fa-lg",
                      "id": "downvote",
                      "onClick": [Function],
                      "style": Object {
                        "float": "right",
                      },
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      undefined,
                    ],
                    "type": "button",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
